frontend/templates/product-list/sections/ProductListChildrenSection.tsx

```diff
@@ -135,17 +138,32 @@ const computeChildVisibility = (
       base: totalCount <= 5 ? 'block' : position < 5 ? 'block' : 'none',
       sm: totalCount <= 8 ? 'block' : position < 7 ? 'block' : 'none',
       md: totalCount <= 9 ? 'block' : position < 8 ? 'block' : 'none',
-      lg: totalCount <= 8 ? 'block' : position < 7 ? 'block' : 'none',
-      xl: totalCount <= 10 ? 'block' : position < 9 ? 'block' : 'none',
+      lg: defaultShowAllChildrenOnLgSizes
+         ? 'block'
+         : totalCount <= 8
+         ? 'block'
+         : position < 7
+         ? 'block'
+         : 'none',
```

Comment:
I find using `||` is easier to read than a nested ternary, but what you have looks correct and I'm not going to block on this.

```suggestion
      lg: defaultShowAllChildrenOnLgSizes || totalCount <= 8 || position < 7
         ? 'block'
         : 'none',
```